x-pack/plugins/discover_log_explorer/public/state_machines/log_explorer_profile/src/state_machine.ts

```diff
@@ -53,41 +60,98 @@ export const createPureLogExplorerProfileStateMachine = (
           invoke: {
             src: 'createDataView',
             onDone: {
-              target: 'initialized',
+              target: 'initializingControlPanels',
             },
             onError: {
               target: 'initialized',
               actions: ['notifyCreateDataViewFailed'],
             },
           },
         },
+        initializingControlPanels: {
+          invoke: {
+            src: 'initializeControlPanels',
+            onDone: {
+              target: 'initialized',
+              actions: ['storeControlPanels'],
+            },
+            onError: {
+              target: 'initialized',
+            },
+          },
+        },
         initialized: {
-          initial: 'idle',
+          type: 'parallel',
           states: {
-            idle: {
-              invoke: {
-                src: 'listenUrlChange',
-              },
-              on: {
-                UPDATE_DATASET_SELECTION: {
-                  target: 'updatingDataView',
-                  actions: ['storeDatasetSelection'],
+            datasetSelection: {
+              initial: 'idle',
+              states: {
+                idle: {
+                  invoke: {
+                    src: 'listenUrlChange',
+                  },
+                  on: {
+                    UPDATE_DATASET_SELECTION: {
+                      target: 'updatingDataView',
+                      actions: ['storeDatasetSelection'],
+                    },
+                    DATASET_SELECTION_RESTORE_FAILURE: {
+                      target: 'updatingDataView',
+                      actions: ['notifyDatasetSelectionRestoreFailed'],
+                    },
+                  },
                 },
-                DATASET_SELECTION_RESTORE_FAILURE: {
-                  target: 'updatingDataView',
-                  actions: ['notifyDatasetSelectionRestoreFailed'],
+                updatingDataView: {
+                  invoke: {
+                    src: 'createDataView',
+                    onDone: {
+                      target: 'idle',
+                      actions: raise('UPDATED_DATA_VIEW'),
```

Comment:
**nit:** we can move it into an action listed among the others in the config object:
```
actions: ['notifyDataViewUpdate']
...
actions: {
 // ...
  notifyDataViewUpdate: raise('DATA_VIEW_UPDATED')
}
```